csp/impl/types/common_definitions.py

```diff
@@ -94,9 +124,34 @@ def __new__(cls, typ, shape: Optional[Union[List, int, str]] = None, shape_of: O
             # if shape is required, it will be enforced in the parser
             kwargs["shape"] = None
             kwargs["shape_func"] = None
+
         return type("OutputBasket", (OutputBasket,), kwargs)
 
 
+# Add core schema to OutputBasket
+def __get_pydantic_core_schema__(cls, source_type, handler):
```

Comment:
any reason not to define this as @classmethod directly on the class?

Reply:
The first time I tried this, I couldn't get it working properly with __new__. Let me try again though to confirm.

Reply:
Fixed, not sure why I didn't just do this the first time like I do everywhere else.